screens/dashboard/components/upcoming.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import moment from 'moment';
+
+// import global actions
+import * as actions from '../actions';
+
+// import global styles
+// @TODO: move the global styles from this screen into ../../styles/common 
+import commonStyles from '../../../styles/common';
+const styles = StyleSheet.create(commonStyles);
+
+type Props = {
+  actions: Object,
+  profile: Object,
+  navigation: Object
+};
+
+
+export class Upcoming extends Component<Props> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+    this.icons = {
+      'open': 'angle-down',
+      'close': 'angle-up'
+    };
+  }
+
+
+  showUpcoming(firstOrRemaining) {
+
+    let upcomingArray = ((this.props.profile || {}).upcomingArray || {});
+
+    let allUpcomingEvents = [];
+    for (i = 0; i < upcomingArray.length; i++) {
```

Comment:
the variable `i` appears to be implicitly declared.   Change it so it's explicit.
Also, this loop is inefficient because the `upcoingArray.length` is recalculated each time.

Also, as a matter of good style, don't use the ++ or -- operator.  It's a common source of bugs.  

I will quote the great and powerful Douglas Crockford from  his book "JavaScript the Good Parts"

```The ++ (increment) and -- (decrement) operators have been known to contribute to bad code by encouraging excessive trickiness. They are second only to faulty architecture in enabling to viruses and other security menaces. There is a plusplus option [in JSLint] that prohibits the use of these operators.```


This would be better:

for (let i = 0; len = upcomingArray.length; i < len; i +=1) {

Reply:
`      const upcomingMapped = upcomingSorted.map((event, i) => {
`

Reply:
This is now ` const upcomingMapped = upcomingSorted.map((event, i) => {`